### DISCUSSION TITLE
excessive number of log entries in syslog

### DISCUSSION THREAD

**Discussion by ostapkostyk** [k3s-io/k3s#9537] [Q&A]
Hi all,
The syslog raises very quickly on all nodes in my cluster. Every second a dozen of info and warning level entries from k3s are added, see example in attached file
[syslog2.txt](https://github.com/k3s-io/k3s/files/14362171/syslog2.txt)
This pollutes it heavily and it grows to gigabyte range in one week.

I have a suspicion that the reason is a bad latency (over 100ms), still not sure is it the reason.
The cluster is "hybrid", master nodes are not in one local network, but connected through the Internet using wire-guard protocol. One node is located in quite a remote location, so the average ping is 150ms. Therefore, I configured embedded ETCD with  --heartbeat-interval=500 and --election-timeout=5000 and to my point of view the cluster works stable enough, I don't see any issues except fast-growing syslog.
I tried to configure warning threshold level and log level for etcd to suppress those info and warning messages, but it doesn't work.
Here is what I did:
In ```/etc/rancher/k3s/config.yaml``` I added following lines:
```yaml
etcd-arg:
  - 'experimental-warning-apply-duration: 300ms'
  - 'log-level: error'
```
and then restarted k3s with ```sudo systemctl restart k3s```. It restarted successfully.
But if look into ```/var/lib/rancher/k3s/server/db/etcd/config``` - above flags didn't appear there. Overall, seems no effect on syslog. Still info and error messages about timing over 100ms (still under 300ms).

So, could anybody explain the reason of those info and warning entries in syslog? Is it because of high latency between etcd nodes? If yes, then how to suppress it (taking into account that in this case latency over 100ms is acceptable)?

**Comment by brandond** [comment]
> ```yaml
> etcd-arg:
>   - 'experimental-warning-apply-duration: 300ms'
>   - 'log-level: error'
> ```

that is not how CLI args work. Reference https://etcd.io/docs/v3.5/op-guide/configuration/.

Did you want:
```yaml
etcd-arg:
  - '--experimental-warning-apply-duration=300ms'
  - '--log-level=error'
```

> The cluster is "hybrid", master nodes are not in one local network, but connected through the Internet using wire-guard protocol. One node is located in quite a remote location, so the average ping is 150ms.

Note that this is technically not a supported configuration; we intentionally use node internal IPs for etcd member addresses, with the expectation that etcd nodes will be on a common high-speed, low-latency network.

**Reply by ostapkostyk** [reply]
Thanks for the tip! I didn't know how to apply these flags correctly and still am not sure.
If I apply as you suggested:
```
etcd-arg:
  - '--experimental-warning-apply-duration=300ms'
  - '--log-level=error'
```
k3s doesn't start, complaints on syntax. On ```sudo journalctl -xeu k3s.service``` I get 
```
Failed to reconcile with temporary etcd: error unmarshaling JSON: while decoding JSON: json: cannot unmarshal string into Go struct field configYAML.experimental-warning-apply-duration of type time.Duration"
```

Then I've changed syntax to:
```
etcd-arg:
  - '--experimental-warning-apply-duration: 300ms'
  - '--log-level: error'
```
In this case, k3s starts but log level seems to be unchganged, I see info and warning messages.

If I change to a mixed syntax:
```
etcd-arg:
  - '--experimental-warning-apply-duration: 300ms'
  - '--log-level=error'
```
then info and warn level messages dissappear from log, k3s starts but in the system log there are messages about continuous restarts of LightweightKubernetes and ```k3s flag provided but not defined: -etcd-s3```, however backup to s3 actually works.
Then, it's interesting that on other nodes this configuration works and k3s doesn't complain like above.
I'm lost.

The full config file:
```
node-name: c-1
etcd-s3: true
etcd-s3-bucket: HJkfgdf....
etcd-s3-region: eu
etcd-s3-endpoint: s3.eu-cen......com
etcd-s3-access-key: 003.....2
etcd-s3-secret-key: K0......U
etcd-snapshot-schedule-cron: "0 */24 * * *"
etcd-snapshot-retention: 25
etcd-arg:
  - '--experimental-warning-apply-duration: 300ms'
  - '--log-level=error'
```

Regarding k3s is not designed to work with high latency. I got it. Will think about the migration for some nodes.
